backend/daemon/daemon_e2e_test.go

```diff
@@ -30,56 +32,76 @@ func TestAPIGetRemotePublication(t *testing.T) {
 	ctx := context.Background()
 
 	// Carol will be the DHT server
-	carol := makeTestApp(t, "carol", makeTestConfig(t), true)
+	dhtProvider := makeTestApp(t, "carol", makeTestConfig(t), true)
 
-	var alice *App
-	{
-		cfg := makeTestConfig(t)
-		cfg.P2P.BootstrapPeers = carol.Net.MustGet().Libp2p().AddrsFull()
-		alice = makeTestApp(t, "alice", cfg, true)
-	}
+	requester, publishedDocument, _ := makeRemotePublication(t, ctx, dhtProvider)
 
-	var bob *App
-	{
-		cfg := makeTestConfig(t)
-		cfg.P2P.BootstrapPeers = carol.Net.MustGet().Libp2p().AddrsFull()
-		bob = makeTestApp(t, "bob", cfg, true)
-	}
+	remotePublication, err := requester.RPC.Documents.GetPublication(ctx, &documents.GetPublicationRequest{DocumentId: publishedDocument.Document.Id})
+	require.NoError(t, err)
+	testutil.ProtoEqual(t, publishedDocument, remotePublication, "remote publication doesn't match")
+}
 
-	// Make sure bob does't know anything about alice.
-	require.NoError(t, bob.Net.MustGet().Libp2p().Network().ClosePeer(alice.Repo.Device().ID()))
-	bob.Net.MustGet().Libp2p().Peerstore().RemovePeer(alice.Repo.Device().ID())
+func TestGateway(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
 
-	draft, err := alice.RPC.Documents.CreateDraft(ctx, &documents.CreateDraftRequest{})
-	require.NoError(t, err)
+	gwConf := makeTestConfig(t)
+	gwConf.P2P.DisableListing = true
 
-	updated, err := alice.RPC.Documents.UpdateDraftV2(ctx, &documents.UpdateDraftRequestV2{
-		DocumentId: draft.Id,
-		Changes: []*documents.DocumentChange{
-			{Op: &documents.DocumentChange_SetTitle{SetTitle: "My new document title"}},
-			{Op: &documents.DocumentChange_SetSubtitle{SetSubtitle: "This is my document's abstract"}},
-			{Op: &documents.DocumentChange_MoveBlock_{MoveBlock: &documents.DocumentChange_MoveBlock{BlockId: "b1"}}},
-			{Op: &documents.DocumentChange_ReplaceBlock{ReplaceBlock: &documents.Block{
-				Id:   "b1",
-				Type: "statement",
-				Text: "Hello world!",
-			}}},
-		},
+	gwConf.Identity.DeviceKeyPath = ("./device.key")
```

Comment:
This will write the file into the source tree. Avoid doing so. Use `ioutil.TempFile` or something to get a temporary file name. If the test fails in some weird way, the file will not pollute the source tree.

Reply:
I thought about it and I was going to to do it manually I know in linux we had a `/tmp` directory but was not sure if that would work on other OS. Didn't know about `ioutil.TempFile`  great!